src/components/ExcalidrawView.tsx

```diff
@@ -13,65 +12,172 @@ import {
+        excalidrawRef.current.getAppState(),
+      );
+      excalidrawRef.current.updateScene({
+        elements: reconciledElements,
+        commitToHistory: false,
+      });
+      setLocalElements(reconciledElements);
+    } else {
+      setLocalElements(newElements);
+    }
+    setIdState(stateAppData.id);
+    setAppState({
+      ...stateAppData.data.appState,
+      collaborators: new Map<string, never>(),
+    });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [appDataArray]);
 
-  const debouncedPatch = useRef(
-    debounce((elements, state) => {
-      if (excalidrawRef.current?.ready && !state.isLoading) {
-        patchAppData({
-          data: { elements, state },
-          id,
-        });
-      }
-    }, DEBOUNCE_VALUE),
+  const saveElements = (
+    elements: readonly ExcalidrawElement[],
+    id: string,
+  ): void => {
+    if (id.length > 0) {
+      patchAppData({
+        id,
+        data: {
+          elements,
+        },
+      });
+    } else {
+      postAppData({
+        ...DEFAULT_EXCALIDRAW_ELEMENTS_APP_DATA,
+        data: {
+          elements,
+        },
+      });
+    }
+  };
+
+  const saveState = (newAppState: AppState, id: string): void => {
+    if (id.length > 0) {
+      patchAppData({
+        id,
+        data: {
+          appState: newAppState,
+        },
+      });
+    } else {
+      postAppData({
+        ...DEFAULT_EXCALIDRAW_STATE_APP_DATA,
+        data: {
+          appState: newAppState,
+        },
+      });
+    }
+  };
+
+  const debouncedSaveElements = useRef(
+    debounce(saveElements, DEBOUNCE_SAVE_ELEMENTS),
   ).current;
 
-  function handleChange(
+  const compareAndSaveElements = (
     elements: readonly ExcalidrawElement[],
-    state: AppState,
-  ): void {
-    // eslint-disable-next-line no-console
-    console.log('handling changes');
-    debouncedPatch(elements, state);
-  }
-
-  useEffect(
-    () => () => {
-      debouncedPatch.cancel();
-    },
-    [debouncedPatch],
-  );
+    prevElements: readonly ExcalidrawElement[],
+    id: string,
+  ): void => {
+    if (!isEqual(elements, prevElements)) {
+      debouncedSaveElements(elements, id);
+    }
+  };
+
+  const debouncedCompareElements = useRef(
+    debounce(compareAndSaveElements, DEBOUNCE_COMPARE_ELEMENTS),
+  ).current;
+
+  const debouncedSaveState = useRef(
+    debounce(saveState, DEBOUNCE_SAVE_STATE),
+  ).current;
+
+  const handleChange = (
+    elements: readonly ExcalidrawElement[],
+    newAppState: AppState,
+  ): void => {
+    const { isResizing, isRotating, isLoading } = newAppState;
+    if (
+      !(isLoading || isResizing || isRotating) &&
+      typeof localElements !== 'undefined'
+    ) {
+      debouncedCompareElements(elements, localElements, idElements);
+      debouncedSaveState(newAppState, idState);
+    } else {
+      debouncedCompareElements.cancel();
+      debouncedSaveElements.cancel();
+      debouncedSaveState.cancel();
+    }
```

Comment:
Can you quickly explain in a comment the context in which this is executed and what it accomplishes ?